### DISCUSSION TITLE
Raspberry Pi - GPIO / I2C

### DISCUSSION THREAD

**Discussion by piotruscilko** [hassio-addons/addon-node-red#1386] [Q&A] [answered]
Hi, I need nodes based on I2C, and GPIO comunication in node-red addon due to remove GPIO components from HASS core:

- node-red-contrib-dht-sensor (I have DHT22 sensor connected to raspberry GPIO16) - node-dht-sensor installed is reqired, but I cannot install it using standard command i HASS OS: answer for comand: "npm install node-dht-sensor" is: "-bash: npm: command not found" how can I install it to go forward?

- node-red-contrib-mcp23017 (I2C bus expansion board connected) - it require also to install: node.js and node-mcp23017 (the same situation as above),

- node-red-contrib-bme280-rpi (I2C bus expansion board connected) - result of run node: "Sensor Init failed ->Error: EPERM: operation not permitted, open '/dev/i2c-1'", how can I give it permissions?

Thanks in advance

**Comment by frenck** [comment, marked as answer]
This add-on has removed support for GPIO as well. If you need those nodes, this add-on isn't for you.

Sorry!

../Frenck

**Reply by Waterbrain** [reply]
And if?
HassOS I2C Configurator


<img width="1048" alt="Captura de Pantalla 2023-11-16 a las 13 09 00" src="https://github.com/hassio-addons/addon-node-red/assets/5563436/10d4363b-6a4b-4dd8-9819-3f83b0ecb05e">

In my case

![Captura de Pantalla 2023-11-16 a las 13 50 04](https://github.com/hassio-addons/addon-node-red/assets/5563436/55ffbd96-1fa1-4b55-946a-96a01036be3c)

![Captura de Pantalla 2023-11-16 a las 13 49 48](https://github.com/hassio-addons/addon-node-red/assets/5563436/3a4a124e-988a-43fd-8586-0a7f4adc8ef4)
